source/common/http/header_map_impl.h

```diff
@@ -130,8 +134,15 @@ class HeaderMapImpl : public HeaderMap {
   /**
    * List of HeaderEntryImpl that keeps the pseudo headers (key starting with ':') in the front
    * of the list (as required by nghttp2) and otherwise maintains insertion order.
+   *
+   * Note: the internal iterators held in fields make this unsafe to copy and move, since the
+   * reference to end() is not preserved across a move (see Notes in
+   * https://en.cppreference.com/w/cpp/container/list/list). The NonCopyable will supress both copy
+   * and move constructors/assignment.
+   * TODO(htuch): Maybe we want this to movable one day; for now, our header map moves happen on
+   * HeaderMapPtr, so the performance impact should not be evident.
    */
-  class HeaderList {
+  class HeaderList : NonCopyable {
```

Comment:
I think it's fine to block move, but why not just block the implicit move constructor inside `NonCopyable`? I realize it's called `NonCopyable` but I'm pretty sure in all cases where we use that we also don't want move?

Reply:
I don't think this change is needed anymore, but fine to keep.